pkg/controller/defaults/placements.go

```diff
@@ -82,6 +82,22 @@ var (
 					getWeightedPodAffinityTerm(100, "rook-ceph-rgw"),
 				},
 			},
+			NodeAffinity: &corev1.NodeAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: getOcsNodeSelector(),
+				PreferredDuringSchedulingIgnoredDuringExecution: []corev1.PreferredSchedulingTerm{
```

Comment:
Looks like getOcsNodeSelector() only uses the default node selector label. If the StorageCluster.spec has a label selector given by the user then it will not be considered. 

May I suggest that you split getStorageClusterEligibleNodes() to do this?